### ISSUE TITLE
different values for "ci" and "errorbar" when using "sd" for pointplot

### ISSUE THREAD

**Issue by PhillipMaire** [mwaskom/seaborn#3149]
hello I am not sure if this is expected behavior or if I am missing something but I am getting different values, see example code
```

import seaborn as sns 
import pandas as pd
import matplotlib.pyplot as plt
d = {'percent correct':[0.996064, 0.997004, 0.996769, 0.996387, 0.996299, 0.996662], 'label': ['set1', 'set2', 'set1', 'set2', 'set1', 'set2'],
    'ind': [0,0,1,1,2,2]}
df = pd.DataFrame(d)
# the two below are the same
plt.figure()
ax = sns.pointplot(x="label", y="percent correct", data=df, dodge=True, color='k', legend='sadf', errorbar='sd', sort=False)
for k in range(3):
    ax = sns.pointplot(x="label", y="percent correct", data=df.loc[df['ind'] == k], dodge=True, errorbar=None, color='b', sort=False)

plt.figure()
ax = sns.pointplot(x="label", y="percent correct", data=df, dodge=True, color='k', legend='sadf', errorbar='ci', sort=False)
for k in range(3):
    ax = sns.pointplot(x="label", y="percent correct", data=df.loc[df['ind'] == k], dodge=True, errorbar=None, color='b', sort=False)
### below CI is different.     
plt.figure()
ax = sns.pointplot(x="label", y="percent correct", data=df, dodge=True, color='k', legend='sadf', ci='sd', sort=False)
for k in range(3):
    ax = sns.pointplot(x="label", y="percent correct", data=df.loc[df['ind'] == k], dodge=True, errorbar=None, color='b', sort=False)

### below is set SD by hand 
plt.figure()
ax = sns.pointplot(x="label", y="percent correct", data=df, dodge=True, color='k', legend='sadf', ci=None, sort=False)
for k in range(3):
    ax = sns.pointplot(x="label", y="percent correct", data=df.loc[df['ind'] == k], dodge=True, errorbar=None, color='b', sort=False)
tmp1 = [plt.xlim(), plt.ylim()]
h_error = df[df['label']=='set1'].std()['percent correct']
w_error = df[df['label']=='set2'].std()['percent correct']

h_U = df[df['label']=='set1'].mean()['percent correct']
w_U = df[df['label']=='set2'].mean()['percent correct']

plt.errorbar(0, h_U, h_error, color='k')
plt.errorbar(1, w_U, w_error, color='k')
plt.xlim(tmp1[0])
plt.ylim(tmp1[1])
```
note the blue data is there to compare the error bars and is the same for each plot
top 2 are the same
**errorbar='sd'**
![image](https://user-images.githubusercontent.com/19584564/202761474-a20a0da7-af1f-49d0-890e-f349ab10a352.png)
**errorbar='ci'**
![image](https://user-images.githubusercontent.com/19584564/202761495-f02c037a-e79d-47b0-84f9-4c40d680e6f5.png)
using CI gives different results 
**ci='sd'**
![image](https://user-images.githubusercontent.com/19584564/202761511-50e95e51-f26f-4f00-bd7d-18c51a29527b.png)
setting custom SD gives slightly different values too
**custom SD error bars**
![image](https://user-images.githubusercontent.com/19584564/202761533-dcd4ca8e-9128-4980-9158-1559ea7fda84.png)

**Comment by PhillipMaire** [comment]
also on a side note is there a good reason why the order of the plot children (lines dot etc ) are are arranged as they were plotted? likely a matplotlib question but just curious.

**Comment by mwaskom** [comment]
I'm finding this example a bit hard to parse, so here's a simpler one:

```python
sns.pointplot(data=tips, x="time", y="total_bill", errorbar="sd", errwidth=8)
ax = sns.pointplot(data=tips, x="time", y="total_bill", ci="sd", color="C1", linestyles="--", errwidth=3)
for line in ax.lines:
    line.set_solid_capstyle("butt")
```
![image](https://user-images.githubusercontent.com/315810/202819691-ebf6ba41-ae75-4792-b6b4-dc9412519317.png)

What's more:
```
assert np.allclose(ax.lines[1].get_xydata(), ax.lines[4].get_xydata())
```
passes.

So I'm not quite sure what's going on in your case, but I'd suggest trying to simplify the demonstration.

**Comment by mwaskom** [comment]
Actually it's not possible for `ci="sd"` and `errorbar="sd"` to behavior differently since they don't go through different code paths: https://github.com/mwaskom/seaborn/blob/master/seaborn/categorical.py#L2837

**Comment by PhillipMaire** [comment]
I noticed that `ci="sd"` is depreciated and just calls `errorbar="sd"` but they still come out different when using my simple example. 
```
import seaborn as sns 
tips = {'total_bill':[0.996064, 0.997004, 0.996769, 0.996387, 0.996299, 0.996662], 'time': ['set1', 'set2', 'set1', 'set2', 'set1', 'set2'],
    'ind': [0,0,1,1,2,2]}

sns.pointplot(data=tips, x="time", y="total_bill", errorbar="sd", errwidth=8)
ax = sns.pointplot(data=tips, x="time", y="total_bill", ci="sd", color="C1", linestyles="--", errwidth=3)
for line in ax.lines:
    line.set_solid_capstyle("butt")
```
![image](https://user-images.githubusercontent.com/19584564/202880882-7d285c10-7453-4190-a49f-2b4c0c7be28e.png)

**Comment by PhillipMaire** [comment]
also both differ from numpy's SD but I assume because seaborn used bootstrapping for everything??
```
import seaborn as sns 
import pandas as pd
import matplotlib.pyplot as plt

tips = {'total_bill':[0.996064, 0.997004, 0.996769, 0.996387, 0.996299, 0.996662], 'time': ['set1', 'set2', 'set1', 'set2', 'set1', 'set2'],
    'ind': [0,0,1,1,2,2]}
tips = pd.DataFrame(tips)

sns.pointplot(data=tips, x="time", y="total_bill", errorbar="sd", errwidth=8)
ax = sns.pointplot(data=tips, x="time", y="total_bill", ci="sd", color="C1", linestyles="--", errwidth=3)
for line in ax.lines:
    line.set_solid_capstyle("butt")


error_1 = tips[tips['time']=='set1'].std()['total_bill']
error_2 = tips[tips['time']=='set2'].std()['total_bill']

h_U = tips[tips['time']=='set1'].mean()['total_bill']
w_U = tips[tips['time']=='set2'].mean()['total_bill']

plt.errorbar(0, h_U, error_1, color='k', zorder=10)
plt.errorbar(1, w_U, error_2, color='k', zorder=10)
```

<img width="596" alt="image" src="https://user-images.githubusercontent.com/19584564/202881570-48494a1c-fdef-4431-87d6-d0e6803ea2b5.png">

**Comment by mwaskom** [comment]
I copy-pasted your example directly and cannot reproduce it:

<img width="986" alt="image" src="https://user-images.githubusercontent.com/315810/202904052-a89bfc36-6cce-4662-a0e8-14dc8f0cf37b.png">

> but I assume because seaborn used bootstrapping for everything??

There's nothing to bootstrap when computing a standard deviation. Although numpy and pandas do have different defaults for the `ddof` parameter in their `std` functions.

**Comment by mwaskom** [comment]
I also cannot reproduce the more complicated version of the example:

![image](https://user-images.githubusercontent.com/315810/202910316-c73e3f0d-fe3a-4b60-8aa5-0f1d33bc9daf.png)

**Comment by mwaskom** [comment]
Your images are smaller than default matplotlib figures should be. What else are you doing? It's really hard to debug something with only partial information about the operations that someone is doing to get a different result...

**Comment by PhillipMaire** [comment]
very weird perhaps its a version issue?? [here is a colab notebook](https://colab.research.google.com/drive/1JGAQT-05awQNCFJ_KRp45Y-z-YwpTvVs?usp=sharing) showing my example.
for the figure sizes I made them 3 by 3 to reduce the white space but that doesn't change my outcome; I removed it in the colab notebook. I would be very surprised if you get a different result from me when using colab. also thank you for your help.

**Comment by PhillipMaire** [comment]
here is a pic of the current versions of the packages and the operating system for colab 
<img width="951" alt="image" src="https://user-images.githubusercontent.com/19584564/202919416-2cbdcf95-a59e-4ec3-9063-4796e6e6077e.png">

note I also showed this error on a mac so maybe its a unix thing?

**Comment by PhillipMaire** [comment]
ok so I used a windows 10 machine and got the same results with the following versions of the packages

<img width="572" alt="image" src="https://user-images.githubusercontent.com/19584564/202919775-debcc0a7-ba6b-44bc-b73f-baf5adce06cc.png">

**Comment by PhillipMaire** [comment]
ok so I figure it out it was a version issue. 
seaborn 0.11.2 is on colab, when I update to seaborn 0.12.1 to problem is resolved. you can see [my notebook](https://colab.research.google.com/drive/1JGAQT-05awQNCFJ_KRp45Y-z-YwpTvVs?usp=sharing) for full replications of my results. so I guess this is solved, I never thought it would be a version issue with something like this since it is such a basic calculation that it seems more likely to be a plotting issue or something. anyways the default version of seaborn on google colab will have this error I am not sure how important this will be for people doing analysis in the future. I only have 3 values so it could be an issue with only a few datapoints? or maybe its dropping one datapoint is that version? just guessing. I'm not sure if you can encourage colab or update the default version of seaborn but if you could that would be ideal.

**Comment by mwaskom** [comment]
Oh, I understand what happened now. (Please always put the versions that you're using in the initial post).

The `errorbar` parameter [was introduced in version 0.12.0](https://seaborn.pydata.org/whatsnew/v0.12.0.html#more-flexible-errorbars). But prior to 0.12.0, `pointplot` would swallow extra kwargs without warning. So what's happening in your example is that you're actually plotting a 95% CI when you invoke `errorbar` (since it's using the default for the `ci` parameter).

I have no idea how to influence the default version on colab, unfortunately.

**Comment by PhillipMaire** [comment]
> Please always put the versions that you're using in the initial post

yes of course this is my mistake, sorry about that. 

I see, thanks for your help that solves why `errorbar='sd'` version of the plot doesn't match `ci='sd'`. 
but it doesn't explain why `ci='sd'` and my custom standard deviation plots don't match before updating, but does match after updating. This suggests that that standard deviation isn't accurate for this particular function in older seaborn versions. 

for version `0.11.2` I see setting `ci='sd'` is a valid for `pointplot`. `pointplot` calls `_PointPlotter` which contains
```
if ci == "sd":

    estimate = estimator(stat_data)
    sd = np.std(stat_data)
    confint.append((estimate - sd, estimate + sd))
```
so it is weird that it doesn't match, but it is likely to have something to do with `estimate = estimator(stat_data)`

**Comment by mwaskom** [comment]
> This suggests that that standard deviation isn't accurate for this particular function in older seaborn versions.

I think the difference you're seeing is explained by 

> Although numpy and pandas do have different defaults for the ddof parameter in their std functions.

**Comment by PhillipMaire** [comment]
you are correct I just tested it. This was really confusing at first, I appreciate your time/help with it. here is the code the verify that the difference I was seeing was do to pandas vs numpy's different ddof parameters. 
```
import seaborn as sns 
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

tips = {'total_bill':[0.996064, 0.997004, 0.996769, 0.996387, 0.996299, 0.996662], 'time': ['set1', 'set2', 'set1', 'set2', 'set1', 'set2'],
    'ind': [0,0,1,1,2,2]}
tips = pd.DataFrame(tips)

# ddof parameter diff check, black and orange are the same, all good. 
ax = sns.pointplot(data=tips, x="time", y="total_bill", ci="sd", color="C1", linestyles="--", errwidth=3)
for line in ax.lines:
    line.set_solid_capstyle("butt")

# numpy STD and MEAN
error_1 = np.std(tips[tips['time']=='set1']['total_bill'])
error_2 = np.std(tips[tips['time']=='set2']['total_bill'])

h_U = np.mean(tips[tips['time']=='set1']['total_bill'])
w_U = np.mean(tips[tips['time']=='set2']['total_bill'])

plt.errorbar(0, h_U, error_1, color='k', zorder=10)
plt.errorbar(1, w_U, error_2, color='k', zorder=10)

```

**Comment by PhillipMaire** [comment]
you are correct I just tested it. This was really confusing at first, I appreciate your time/help with it. here is the code the verify that the difference I was seeing was do to pandas vs numpy's different ddof parameters. 
```
import seaborn as sns 
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

tips = {'total_bill':[0.996064, 0.997004, 0.996769, 0.996387, 0.996299, 0.996662], 'time': ['set1', 'set2', 'set1', 'set2', 'set1', 'set2'],
    'ind': [0,0,1,1,2,2]}
tips = pd.DataFrame(tips)

# ddof parameter diff check, black and orange are the same, all good. 
ax = sns.pointplot(data=tips, x="time", y="total_bill", ci="sd", color="C1", linestyles="--", errwidth=3)
for line in ax.lines:
    line.set_solid_capstyle("butt")

# numpy STD and MEAN
error_1 = np.std(tips[tips['time']=='set1']['total_bill'])
error_2 = np.std(tips[tips['time']=='set2']['total_bill'])

h_U = np.mean(tips[tips['time']=='set1']['total_bill'])
w_U = np.mean(tips[tips['time']=='set2']['total_bill'])

plt.errorbar(0, h_U, error_1, color='k', zorder=10)
plt.errorbar(1, w_U, error_2, color='k', zorder=10)

```
